gobblin-utility/src/main/java/gobblin/util/SchedulerUtils.java

```diff
@@ -178,6 +210,81 @@ public boolean accept(File file) {
+          throw new RuntimeException("Found more than one .properties file in directory: " + configDirPath);
+        }
+
+        // Load the properties, which may overwrite the same properties defined in the parent or ancestor directories.
+        // Open the inputStream, construct a reader and send to the loader for constructing propertiesConfiguration.
+        PropertiesConfiguration propertiesConfiguration = new PropertiesConfiguration();
+        Path uniqueConfigFilePath = propertiesFilesStatus[0].getPath();
+        try (InputStreamReader inputStreamReader = new InputStreamReader(filesystem.open(uniqueConfigFilePath),
+            Charsets.UTF_8)) {
+          propertiesConfiguration.load(inputStreamReader);
+          rootProps.putAll(ConfigurationConverter.getProperties(propertiesConfiguration));
+        }
+      }
+
+      // Get all non-properties files
+      FileStatus[] nonPropFiles = filesystem.listStatus(configDirPath, NON_PROPERTIES_PATH_FILTER);
+      if (nonPropFiles.length == 0) {
+        return;
+      }
+
+      for (FileStatus nonPropFile : nonPropFiles) {
+        Path configFilePath = nonPropFile.getPath();
+        if (nonPropFile.isDirectory()) {
+          Properties rootPropsCopy = new Properties();
+          rootPropsCopy.putAll(rootProps);
+          loadGenericJobConfigsRecursive(jobConfigs, rootPropsCopy, jobConfigFileExtensions, configFilePath);
+        } else {
+          if (!jobConfigFileExtensions.contains(
+              configFilePath.getName().substring(configFilePath.getName().lastIndexOf('.') + 1).toLowerCase())) {
+            LOGGER.warn("Skipped file " + configFilePath + " that has an unsupported extension");
+            continue;
+          }
+
+          Path doneFilePath = configFilePath.suffix(".done");
+          if (filesystem.exists(doneFilePath)) {
+            LOGGER.info("Skipped job configuration file " + doneFilePath + " for which a .done file exists");
+            continue;
+          }
+
+          Properties jobProps = new Properties();
+          // Put all parent/ancestor properties first
+          jobProps.putAll(rootProps);
+          // Then load the job configuration properties defined in the job configuration file
+          PropertiesConfiguration propertiesConfiguration = new PropertiesConfiguration();
+          try (InputStreamReader inputStreamReader = new InputStreamReader(filesystem.open(configFilePath),
+              Charsets.UTF_8)) {
+            propertiesConfiguration.load(inputStreamReader);
+            jobProps.putAll(ConfigurationConverter.getProperties(propertiesConfiguration));
+            jobProps.setProperty(ConfigurationKeys.JOB_CONFIG_FILE_PATH_KEY, configFilePath.toString());
+            jobConfigs.add(jobProps);
+          }
+        }
+      }
+    }
+  }
+
+  /**
    * Recursively load job configuration files under the given directory.
    */
   private static void loadJobConfigsRecursive(List<Properties> jobConfigs, Properties rootProps,
```

Comment:
After first PR got merged, I can migrate related part in #1066 and create a new PR. That one will remove all the deprecate method. Since #1066 seems more not that trivial as replacing File with Path, I will focus on that then.